src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -565,20 +590,49 @@ struct DecimalBinaryOperation
         {
             if (scale_a != 1)
             {
-                for (size_t i = 0; i < size; ++i)
-                    c[i] = applyScaled<true>(
-                        static_cast<NativeResultType>(unwrap<op_case, OpCase::LeftConstant>(a, i)),
-                        static_cast<NativeResultType>(unwrap<op_case, OpCase::RightConstant>(b, i)),
-                        scale_a);
+                if constexpr (is_compare)
+                {
+                    for (size_t i = 0; i < size; ++i)
+                        c[i] = applyScaled<true>(
+                            static_cast<NativeResultType>(unwrap<op_case, OpCase::LeftConstant>(a, i)),
+                            static_cast<NativeResultType>(unwrap<op_case, OpCase::RightConstant>(b, i)),
+                            scale_a,
+                            left_nullmap && ((*left_nullmap)[i]),
+                            right_nullmap && ((*right_nullmap)[i]));
```

Comment:
Could we use a lambda func to process most of logic so that the below three part would be more clean?

Reply:
done